lib/contextual_logger.rb

```diff
@@ -43,31 +44,40 @@ def normalize_message(message)
   end
 
   module LoggerMixin
+    include Context
+
     delegate :register_secret, :register_secret_regex, to: :redactor
 
+    def global_context
+      @global_context ||= {}.freeze
+    end
+
     def global_context=(context)
-      Context::Handler.new(context).set!
+      @global_context = context.freeze
```

Comment:
@jebentier Per the other comment, I probably should do a `deep_freeze` here?

But also, it occurred to me that this simple assignment won't actually work if there is any `current_context_override` set. The reason is that that override is in fact the merge of some explicit context and the current context at the time... which would always have included the global one.

So if feels like I should `raise ArgumentError` here if there's a `current_context` set. Either that or I should fully implement it where the `current_context_override` just remembers the override bit, and the merging with the global is done separately...but ideally optimized to not be done each time.

Reply:
I went with what we discussed: the first time any context is overridden with `with_context`, that locks the `global_context`. It may not be changed after that.